src/probability/density.lean

```diff
@@ -347,12 +347,32 @@ lemma pdf_to_real_ae_eq {m : measurable_space α}
   (λ x, (s.indicator ((μ s)⁻¹ • (1 : E → ℝ≥0∞)) x).to_real) :=
 filter.eventually_eq.fun_comp hX ennreal.to_real
 
-variables [is_finite_measure ℙ] {X : α → ℝ}
-variables {s : set ℝ} (hms : measurable_set s) (hns : volume s ≠ 0)
+variables {X : α → ℝ} {s : set ℝ} (hms : measurable_set s) (hns : volume s ≠ 0)
 
 include hms hns
 
-lemma mul_pdf_integrable (hcs : is_compact s) (huX : is_uniform X s ℙ) :
+lemma measure_preimage {A : set ℝ} (huX : is_uniform X s ℙ) (hA : measurable_set A)
+  (hnt : volume s ≠ ⊤) :
```

Comment:
I expect this and the below might be provable without some of the assumptions on `s` but I couldn't figure it out

Reply:
I'm guessing its not provable without `hns` and `hnt` since in those cases, `is_uniform` does not give us `has_pdf` and so we can't say anything about the push forward measure.

Reply:
I see - what about `is_probability_measure`?

Reply:
We need `hns` and `hnt` for the same reason (`is_uniform` does not give us any information if `s` has measure 0 or infinity)